File: src/main/java/axirassa/webapp/pages/monitor/CreateMonitor.java

package axirassa.webapp.pages.monitor;

import java.util.Arrays;
import java.util.List;

import org.apache.shiro.authz.annotation.RequiresUser;
import org.apache.tapestry5.ValueEncoder;
import org.apache.tapestry5.annotations.Property;
import org.apache.tapestry5.hibernate.annotations.CommitAfter;
import org.apache.tapestry5.ioc.annotations.Inject;
import org.apache.tapestry5.util.EnumValueEncoder;
import org.hibernate.Session;

import axirassa.dao.UserDAO;
import axirassa.model.MonitorType;
import axirassa.model.PingerEntity;
import axirassa.model.PingerFrequency;
import axirassa.webapp.services.AxirassaSecurityService;
import axirassa.webapp.services.exceptions.AxirassaSecurityException;

@RequiresUser
public class CreateMonitor {
	@Inject
	private Session session;

	@Inject
	private AxirassaSecurityService security;

	@Inject
	private UserDAO userDAO;

	@Property
	private final ValueEncoder<PingerFrequency> frequencyEncoder = new EnumValueEncoder(PingerFrequency.class);

	@Property
	private String url;

	@Property
	private PingerFrequency monitorFrequency;

	@Property
	private final List<PingerFrequency> frequencies = Arrays.asList(PingerFrequency.values());


	@CommitAfter
	public Object onSuccess () throws AxirassaSecurityException {
		// save the pinger
		PingerEntity pinger = new PingerEntity();
		pinger.setUrl(url);
		pinger.setFrequency(monitorFrequency);
		pinger.setUser(security.getUserEntity());
		pinger.setMonitorType(MonitorType.HTTP);
		session.save(pinger);

		return ListMonitor.class;
	}
}

File: src/main/java/axirassa/webapp/pages/monitor/CreateMonitor.scala
package axirassa.webapp.pages.monitor

import axirassa.model.MonitorType
import axirassa.model.PingerEntity
import org.apache.tapestry5.hibernate.annotations.CommitAfter
import java.util.Arrays
import axirassa.model.PingerFrequency
import org.apache.tapestry5.util.EnumValueEncoder
import org.apache.tapestry5.annotations.Property
import axirassa.dao.UserDAO
import axirassa.webapp.services.AxirassaSecurityService
import org.hibernate.Session
import org.apache.tapestry5.ioc.annotations.Inject
import org.apache.shiro.authz.annotation.RequiresUser

@RequiresUser
class CreateMonitor {
	@Inject
	var database : Session = _
	
	@Inject
	var security : AxirassaSecurityService =_
	
	@Inject
	var userDAO : UserDAO = _
	
	@Property
	var url : String = _
	
	@Property
	var monitorFrequency : PingerFrequency = _
	
	@CommitAfter
	def onSuccess = {
	    val pinger = new PingerEntity
	    
	    pinger.setUrl(url)
	    pinger.setFrequency(monitorFrequency.getInterval)
	    pinger.setUser(security.getUserEntity)
//	    pinger.setMonitorType(MonitorType.HTTP)
	    
	    database.save(pinger)
	    
	    classOf[ListMonitor]
	}
}